src/nodes.ts

```diff
@@ -22,31 +23,33 @@ export type PropertyKey = Identifier | Literal;
 export type PropertyValue = AssignmentPattern | BindingIdentifier | BindingPattern | FunctionExpression;
 export type StatementListItem = Declaration | Statement;
 
+type Nullable<T> = T | null;
```

Comment:
Since I don't like disguising types that union in null (because they get lost in the maze with types that don't admit null & you don't want to have to remember which types do and which types don't), I have started using the following naming convention in my own TS2 projects.

`type IdentifierQ = Identifier | null`

The intent is that the letter 'Q' stands for 'question mark' as in the Kleene operator '?' or zero-or-one occurrences of values of the stem type.

Also I don't mind 'S' as another identifier suffix which stands for 'star' as in Kleen '*'. This is useful for array types & other collections which may contain zero-or-more occurrences of said value.

Personally I find this to be a low-noise naming convention; of course everyone has there own stylistic preferences.

Acknowledgments to XPath and XQuery type syntax for the inspiration (also occurrence characters used in RegEx expressions).

Any suggestions for '+', the one-or-more variation?